src/behaviours/Tweened/index.js

```diff
@@ -0,0 +1,82 @@
+import React, { PureComponent } from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import { omit, pick } from 'lodash';
+import uuid from 'uuid';
+import getAbsoluteBoundingRect from '../../utils/getAbsoluteBoundingRect';
+import tween from './tween';
+
+window.Tweens = window.Tweens || [];
+
+function getDisplayName(WrappedComponent) {
+  return WrappedComponent.displayName || WrappedComponent.name || 'Component';
+}
+
+const Tweened = WrappedComponent =>
+  class extends PureComponent {
+    static propTypes = {
+      name: PropTypes.string.isRequired,
+      before: PropTypes.number,
+      duration: PropTypes.number,
+      after: PropTypes.number,
+      uuid: PropTypes.string,
+    };
+
+    static defaultProps = {
+      before: 500,
+      duration: 500,
+      after: 300,
+      uuid: undefined,
+    };
+
+    static displayName = `Tweened(${getDisplayName(WrappedComponent)})`;
+
+    constructor(props) {
+      super(props);
+      this.uuid = this.props.uuid || uuid();
+    }
+
+    componentDidMount() {
+      this.node = ReactDOM.findDOMNode(this); // eslint-disable-line react/no-find-dom-node
+
+      window.Tweens = {
+        ...window.Tweens,
+        [this.props.name]: {
+          ...window.Tweens[this.props.name],
+          [this.uuid]: {
```

Comment:
This looks like it's called at the point of adding a tweened destination. e.g.,  'overview-panel' is added when I click the 'protocol-stack'. But its position info doesn't change before the cached information runs. So if the position changes between route changes, the tween starts from the wrong place.

Steps:
- Click protocol stack
- Change browser viewport width to 50% or so
- Click overview component

Seen: tween starts from the old (cached) position/dimensions of the overview component.

Reply:
I don't mean to nitpick on the tweening behavior; I'm more trying to understand how the tween objects are initialized and cached (and determine if they need to be.)

Reply:
Hmm, I think I will try only registering the elements here, and then getting the positions on tween.

Also, using a redux store for `window.Tweens`, like you suggested above, might make the intent clearer.